### PR TITLE
feat: add fqcnTopicList in api search and filter non fqcnTopicList

### PR DESCRIPTION
**Pull request by azam-ismail**

### CODE REVIEW

File: src/main/java/org/energyweb/ddhub/dto/SearchMessageDTO.java [hunk 1]
```diff
@@ -28,10 +28,17 @@ public class SearchMessageDTO {
 	private String fqcn;
 
 	@NotNull
+	@NotEmpty
```
Comment (kim-energyweb) [inline src/main/java/org/energyweb/ddhub/dto/SearchMessageDTO.java:31]: Why not empty is added here?

Comment (azam-ismail) [reply src/main/java/org/energyweb/ddhub/dto/SearchMessageDTO.java:31]: handle for []

File: src/main/java/org/energyweb/ddhub/Message.java [hunk 2]
```diff
@@ -474,7 +474,7 @@ public Response searchInternal(@Valid @NotNull SearchInternalMessageDTO messageD
     @Authenticated
     public Response search(@Valid @NotNull SearchMessageDTO messageDTO)
             throws InterruptedException, TimeoutException, IOException, JetStreamApiException {
-        topicRepository.validateTopicIds(messageDTO.getTopicId(), true);
```
Comment (kim-energyweb) [inline src/main/java/org/energyweb/ddhub/Message.java:477]: I think this is also needed if topicId is not null or empty.

Comment (azam-ismail) [reply src/main/java/org/energyweb/ddhub/Message.java:477]: that already update to topicRepository.validateTopicIds(messageDTO.getFqcnTopicList(), true);

**Review by kim-energyweb — APPROVED** [review]
LGTM

**Pull request merged**